core/txpool/legacypool/list.go

```diff
@@ -367,11 +353,11 @@ func (l *list) Filter(costLimit *uint256.Int, gasLimit uint64) (types.Transactio
 	if l.costcap.Cmp(costLimit) <= 0 && l.gascap <= gasLimit {
 		return nil, nil
 	}
-	l.costcap = new(uint256.Int).Set(costLimit) // Lower the caps to the thresholds
+	l.costcap.Set(costLimit) // Lower the caps to the thresholds
```

Comment:
It's wrong. This will modify the `l.costcap` which may be referenced by another tx.cost at line339